src/main/java/com/udemy/elearning/services/CheckoutService.java

```diff
@@ -0,0 +1,75 @@
+package com.udemy.elearning.services;
+
+
+import com.udemy.elearning.dto.CheckoutRequest;
+import com.udemy.elearning.models.CardInfo;
+import com.udemy.elearning.models.Checkout;
+import com.udemy.elearning.models.CheckoutCourse;
+import com.udemy.elearning.models.Course;
+import com.udemy.elearning.repository.CardInfoRepository;
+import com.udemy.elearning.repository.CheckoutCourseRepository;
+import com.udemy.elearning.repository.CheckoutRepository;
+import com.udemy.elearning.repository.CourseRepository;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.springframework.stereotype.Service;
+import org.webjars.NotFoundException;
+
+import java.util.List;
+
+@Service
+public class CheckoutService {
+
+    private static final Logger logger = LogManager.getLogger(CheckoutService.class);
+
+    private final CheckoutRepository checkoutRepository;
+    private final CheckoutCourseRepository checkoutCourseRepository;
+    private final CourseRepository courseRepository;
+    private final CardInfoRepository cardInfoRepository;
+
+    public CheckoutService(CheckoutRepository checkoutRepository, CheckoutCourseRepository checkoutCourseRepository, CourseRepository courseRepository, CardInfoRepository cardInfoRepository) {
+        this.checkoutRepository = checkoutRepository;
+        this.checkoutCourseRepository = checkoutCourseRepository;
+        this.courseRepository = courseRepository;
+        this.cardInfoRepository = cardInfoRepository;
+    }
+
+    public Checkout create(CheckoutRequest checkoutRequest){
+        CardInfo cardInfo = new CardInfo();
+        cardInfo.setCardNumber(checkoutRequest.getCardNumber());
+        cardInfo.setCardHolderName(checkoutRequest.getCardHolderName());
+        cardInfo.setCardType(checkoutRequest.getCardType());
+        cardInfo.setCardExpiry(checkoutRequest.getCardExpiry());
+        cardInfo.setCardCVC(checkoutRequest.getCardCVC());
+        CardInfo cardInfo1 = cardInfoRepository.save(cardInfo);
+
+        Checkout checkout = new Checkout();
+        checkout.setUserId(checkoutRequest.getUserId());
+        checkout.setCardInfoId(cardInfo1.getId());
+        checkout.setTotalAmount(checkoutRequest.getTotalAmount());
+        Checkout checkout1 = checkoutRepository.save(checkout);
+
+
+
+        for (long courseId: checkoutRequest.getCourseId()){
+            CheckoutCourse checkoutCourse = new CheckoutCourse();
+            checkoutCourse.setCourseId(courseId);
+            checkoutCourse.setCheckoutId(courseId);
+            Course course = courseRepository.findById(courseId).orElseThrow(()->new NotFoundException("Course not found"));
+            checkoutCourse.setPrice(course.getPrice());
```

Comment:
when you have 
 for (long courseId: checkoutRequest.getCourseId()){
            CheckoutCourse checkoutCourse = new CheckoutCourse();
            checkoutCourse.setCourseId(courseId);
            checkoutCourse.setCheckoutId(courseId);
            Course course = courseRepository.findById(courseId).orElseThrow(()->new NotFoundException("Course not found"));
            checkoutCourse.setPrice(course.getPrice());
            CheckoutCourse checkoutCourseResult = checkoutCourseRepository.save(checkoutCourse);
        }


please add  @Transactional annotation on top of Checkout method

Reply:
What is purpose with @transactional bong?